### PR TITLE
Improve diagnostic message if unexpected tokens are found at the start of a layout node

### PR DESCRIPTION
**Pull request by ahoppen**
This helps us produce better error messages once we implement recovery in front of declarations.

Unfortunately, we don’t have a test case that exercises the new paths yet.

### CODE REVIEW

**Comment by ahoppen** [comment]
@swift-ci Please test

**Review by CodaFi — APPROVED** [review]
Doug and I added some more tests that are messing with you here :)

**Comment by ahoppen** [comment]
@swift-ci Please test

File: Sources/SwiftParser/Diagnostics/ParserDiagnosticMessages.swift [hunk 1]
```diff
@@ -16,12 +16,20 @@ import SwiftSyntax
 let diagnosticDomain: String = "SwiftParser"
 
 extension SyntaxProtocol {
-  var nodeTypeNameForDiagnostics: String? {
+  /// Return a name of this syntax node that can be used to describe it in
+  /// diagnostics.
+  /// Nodes that mostly exist for the syntax tree's structure and don't have a
+  /// correspondence in the source code that's meeingful to the user by default
+  /// use the name of the parent node that encloses it. Pass `false` to `inherit`
+  /// to prevent this name inheritance.
```
Comment (bnbarham) [inline Sources/SwiftParser/Diagnostics/ParserDiagnosticMessages.swift:24]: I know this is already merged, but FWIW there's `meeingful` + that sentence is a little hard to parse. Maybe:
> Return the name of this syntax node to be used to describe it in diagnostics. If the node itself has no diagnostic name (eg. it exists for the syntax tree's structure), use the name of the enclosing parent node by default. Pass `inherit: false` to prevent this behavior, in which case `nil` will be returned instead.

Comment (ahoppen) [reply Sources/SwiftParser/Diagnostics/ParserDiagnosticMessages.swift:24]: https://github.com/apple/swift-syntax/pull/760
